var/spack/repos/builtin/packages/py-greenlet/package.py

```diff
@@ -0,0 +1,37 @@
+##############################################################################
+# Copyright (c) 2013-2017, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/spack/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+from spack import *
+
+
+class PyGreenlet(PythonPackage):
+    """Lightweight in-process concurrent programming"""
+
+    homepage = "https://github.com/python-greenlet/greenlet"
+    url      = "https://pypi.io/packages/source/g/greenlet/greenlet-0.4.13.tar.gz"
+
+    version('0.4.13', '6e0b9dd5385f81d478451ec8ed1d62b3')
+
+    # Although setup.py tries to import it and only falls back to distutils
+    # setup.py does not recognize --single-version-externally-managed
+    # depends_on('py-setuptools', type='build')
```

Comment:
That's because of this block:
```python
try:                                                                            
    if not (sys.modules.get("setuptools")                                       
            or "develop" in sys.argv                                            
            or "upload" in sys.argv                                             
            or "bdist_egg" in sys.argv                                          
            or "bdist_wheel" in sys.argv                                        
            or "test" in sys.argv):                                             
        raise ImportError()                                                     
    from setuptools import setup, Extension                                     
    setuptools_args = dict(test_suite='tests.test_collector', zip_safe=False)   
except ImportError:                                                             
    from distutils.core import setup, Extension                                 
    setuptools_args = dict()     
```
It only uses `setuptools` for those 5 `setup.py` args and uses `distutils` for everything else.

Reply:
Should I simply drop the `setuptools` build dependency then?

Reply:
Yeah